src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -833,6 +833,14 @@
           "used dead by replacing them with a Halt node. Turning this off " \
           "could corrupt the graph in rare cases and should be used with "  \
           "care.")                                                          \
+                                                                            \
+  product(bool, ShortRunningLongLoop, true, DIAGNOSTIC,                     \
+          "long counted loop/long range checks: don't create loop nest if " \
+          "loop runs for small enough number of iterations.")               \
```

Comment:
It sounds like we are doing this:
Disable an exception, which disables an optimization.

This double negation can be a little confusing / ambiguous.

I wonder if we should instead have a limit here, which we can move to 0, or higher.
That would allow us to benchmark with different levels more easily too.

Reply:
It could be named `ShortRunningLongLoopIterationLimit`.

Reply:
It could be helpful if you say what we still do:
- do we still convert to int-loop?
- Do we still strip-mine?